OutOfSchool/OutOfSchool.WebApi.Tests/Services/ApplicationServiceTests.cs

```diff
@@ -439,6 +442,42 @@ public async Task UpdateApplication_WhenIdIsValid_ShouldReturnApplication()
         AssertApplicationsDTOsAreEqual(expected, result);
     }
 
+    [Test]
+    public async Task UpdateApplication_WhenIdIsValidAndNeedUpdateWorkshopStatus_ShouldReturnApplication()
+    {
+        // Arrange
+        var id = new Guid("1745d16a-6181-43d7-97d0-a1d6cc34a8bd");
+        var entity = WithApplication(id);
+        var changedEntity = WithApplication(id, ApplicationStatus.Approved);
+        var userId = Guid.NewGuid().ToString();
+        var workshop = WithWorkshop(new Guid("0083633f-4e5b-4c09-a89d-52d8a9b89cdb"));
+
+        var applicationsMock = WithApplicationsList().AsQueryable().BuildMock();
+
+        applicationRepositoryMock.Setup(r => r.Get(
```

Comment:
Are you sure we need this setup? I guess this setup isn't matched for this test. Please check it using applicationRepositoryMock.VerifyAll(); in assert section

Reply:
1. I read that using VerifyAll() is a bad idea.
For example:
https://hamidmosalla.com/2019/02/17/moq-whats-wrong-with-using-verifyall/
2. Using VerifyAll does not indicate that we can abandon setup methods.

Reply:
You are right. setup the Get method is not needed. It does not run in scope of Update method.